app/services/search_service.rb

```diff
@@ -12,6 +12,8 @@ def call(query, account, limit, options = {})
     default_results.tap do |results|
       if url_query?
         results.merge!(url_resource_results) unless url_resource.nil?
+        results[:accounts].reject! { |item| item.blocking?(@account) }
+        results[:statuses].reject! { |status| StatusFilter.new(status, @account).filtered? }
```

Comment:
Note: We run StatusFilter on ElasticSearch results too, though I don't remember if it's in this class or the controller. Might be worth considering just running it once before returning results, instead of in each branch separately.

Reply:
It's done in each branch separately, we could refactor it, but we wouldn't gain much.